omegaconf/grammar_visitor.py

```diff
@@ -168,37 +168,22 @@ def visitInterpolationNode(
     def visitInterpolationResolver(
         self, ctx: OmegaConfGrammarParser.InterpolationResolverContext
     ) -> Optional["Node"]:
-        from ._utils import _get_value
 
-        # INTER_OPEN (interpolation | ID) COLON sequence? BRACE_CLOSE;
-        resolver_name = None
+        # INTER_OPEN resolverName COLON sequence? BRACE_CLOSE
+        assert 4 <= ctx.getChildCount() <= 5
+
+        resolver_name = self.visit(ctx.getChild(1))
+        maybe_seq = ctx.getChild(3)
         args = []
         args_str = []
-        for child in ctx.getChildren():
-            if (
-                isinstance(child, TerminalNode)
-                and child.symbol.type == OmegaConfGrammarLexer.ID
-            ):
-                assert resolver_name is None
-                resolver_name = child.symbol.text
-            elif isinstance(child, OmegaConfGrammarParser.InterpolationContext):
-                assert resolver_name is None
-                resolver_name = _get_value(self.visitInterpolation(child))
-                if not isinstance(resolver_name, str):
-                    raise GrammarParseError(
-                        f"The name of a resolver must be a string, but the interpolation "
-                        f"{child.getText()} resolved to `{resolver_name}` which is of type "
-                        f"{type(resolver_name)}"
-                    )
-            elif isinstance(child, OmegaConfGrammarParser.SequenceContext):
-                assert resolver_name is not None
-                for val, txt in self.visitSequence(child):
-                    args.append(val)
-                    args_str.append(txt)
-            else:
-                assert isinstance(child, TerminalNode)
+        if isinstance(maybe_seq, TerminalNode):  # means there are no args
+            assert maybe_seq.symbol.type == OmegaConfGrammarLexer.BRACE_CLOSE
+        else:
+            assert isinstance(maybe_seq, OmegaConfGrammarParser.SequenceContext)
+            for val, txt in self.visitSequence(maybe_seq):
+                args.append(val)
+                args_str.append(txt)
```

Comment:
can you explain the change here?

Reply:
In the previous version, `visitInterpolationResolver()` was computing everything: the resolver name (which could be either an `ID` or an interpolation), then the sequence of arguments.

In the new version, to make the grammar more readable, I moved the parsing of the resolver name to a new parser rule, and the first two `if`/`elif` statements have thus been moved to a new visitor method `visitResolverName()`. The only part that's left in `visitInterpolationResolver()` is the computation of the list of arguments.